server/src/auth/auth.service.ts

```diff
@@ -0,0 +1,48 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { StaffService } from 'src/models/staff/staff.service';
+import { BcryptService } from './bcrypt.service';
+import { JwtService } from '@nestjs/jwt';
+
+//TODO: return roles associated as well when that relation has been established
+const SELECT = {
+    id: true,
+    firstName: true,
+    lastName: true,
+    email: true,
+    authKey: true,
+    gender: true,
+    isSuperUser: true,
+}
+
+@Injectable()
+export class AuthService {
+    constructor(
+        private staffService: StaffService,
+        private jwtService: JwtService,
+        private bcrypt: BcryptService
+    ) { }
+    async validateUser(email: string, password: string) {
+        const user = await this.getDetails({ email }, SELECT)
+        if (user?.authKey && this.verifyPassword(password, user.authKey)) {
```

Comment:
`bcrypt.verify` returns a promise, from what I am seeing you are passing the promise directly to the if condition, this will result in the condition to always be truthy. You must await the results first:

```ts
const valid = await this.verifyPassword(pass, hash)

// or

if (await this.verifyPassword(pass, hash)) {
}
```

We should add eslint rule to prevent promise being placed in if conditions. @KonomiAI/konomi-core-team